File: app/src/main/java/org/deiverbum/app/model/PaterOpus.java
package org.deiverbum.app.model;

import androidx.room.Ignore;

import java.text.MessageFormat;

public class PaterOpus {
    public Integer opusID;
    public String opusName;
    public String liturgyName;
    public String subTitle;
    public String opusDate;
    public String volume;
    public String editorial;
    public String city;
    public String opusYear;
    public Integer paterFK;
    public Integer typeFK;
    public Integer collectionFK;

    @Ignore
    private Pater pater;

    public PaterOpus() {
    }

    public String getOpusName() {
        return this.opusName;
    }

    public void setOpusName(String opusName) {
        this.opusName = opusName;
    }

    public Pater getPater() {
        return this.pater;
    }

    public void setPater(Pater pater) {
        this.pater = pater;
    }

    public String getOpusForView() {
        return liturgyName!=null && !liturgyName.equals("") ? liturgyName : MessageFormat.format("{0}, {1}",pater.getPater(), opusName);
    }
}

File: app/src/main/java/org/deiverbum/app/model/PaterOpus.kt
package org.deiverbum.app.model

import androidx.room.Ignore
import java.text.MessageFormat

class PaterOpus {
    var opusID: Int? = null
    @JvmField
    var opusName: String? = null
    @JvmField
    var liturgyName: String? = null
    var subTitle: String? = null
    var opusDate: String? = null
    var volume: String? = null
    var editorial: String? = null
    var city: String? = null
    var opusYear: String? = null
    var paterFK: Int? = null
    var typeFK: Int? = null
    var collectionFK: Int? = null

    @Ignore
    var pater: Pater? = null
    val opusForView: String
        get() = if (liturgyName != null && liturgyName != "") liturgyName!! else MessageFormat.format(
            "{0}, {1}",
            pater?.pater,
            opusName
        )

    val paterForView: String
        get() = if (pater?.liturgyName!=null && pater?.liturgyName!="") pater?.liturgyName!! else pater?.pater!!

    val singleName: String
        get() = if (opusName != null) opusName!! else ""
}